File: DIP/DIP_Lab_406/Lab/MatLab/Lab-04/MedianFilter.m
image = imread('nature.jpeg');
if size(image, 3) == 3
    grayImage = rgb2gray(image);
else
    grayImage = image;
end
filterSize = 3;   
filteredImage = medfilt2(grayImage, [filterSize filterSize]);
figure;
subplot(1, 2, 1);
imshow(grayImage);
title('Original Image');
subplot(1, 2, 2);
imshow(filteredImage);
title('Median Filtered Image');

File: DIP/DIP_Lab_406/Lab/353_lab_04/Lab_04_python/MedianFilter.py
import cv2
import numpy as np
from matplotlib import pyplot as plt
image = cv2.imread('nature.jpeg')
image_rgb = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
median_filtered_image = cv2.medianBlur(image_rgb, 7)
plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.imshow(image_rgb)
plt.title('Original Image')
plt.axis('off')
plt.subplot(1, 2, 2)
plt.imshow(median_filtered_image)
plt.title('Median Filtered Image')
plt.axis('off')
plt.show()
